### ISSUE TITLE
Error when subclassing for stock example: Expected D2 of index to be 2 got 3 at position 1

### ISSUE THREAD

**Issue by Guzzii** [tensorflow/tensorflow#31733]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
Yes. The code is derived from one of the integration test cases.
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Stock example in Google Colab https://colab.research.google.com/github/tensorflow/docs/blob/r2.0rc/site/en/r2/tutorials/keras/feature_columns.ipynb
- TensorFlow version (use command below):
tf.version.GIT_VERSION = v2.0.0-beta0-16-g1d91213fe7
tf.version.VERSION = 2.0.0-beta1
- Python version:
3.6

**Describe the current behavior**
The following `feature_column` creates the issue when subclassing `keras.models.Model`, which works alright with `sequential` API.
```python
crossed_feature = feature_column.crossed_column([age_buckets, thal], hash_bucket_size=1000)
crossed_feature = feature_column.indicator_column(crossed_feature)
```

**Describe the expected behavior**
Not failing with `feature_column.crossed_column` when subclassing. The integration test here, https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/keras/engine/feature_columns_integration_test.py#L161-L183, didn't catch this problem.

**Code to reproduce the issue**
Running the following at the end of the notebook will reproduce the error.
```python
class FeatureColumnDsModel(tf.keras.models.Model):
    def __init__(self, feature_columns, **kwargs):
        super().__init__(name=None, **kwargs)
        self._input_layer = tf.keras.layers.DenseFeatures(feature_columns)
        self._dense1 = tf.keras.layers.Dense(128, activation='relu')
        self._dense2 = tf.keras.layers.Dense(128, activation='relu')
        self._output_layer = tf.keras.layers.Dense(1, activation='sigmoid')
        
    def call(self, features):
        x = self._input_layer(features)
        x = self._dense1(x)
        x = self._dense2(x)
        y = self._output_layer(x)
        return y
    
model = FeatureColumnDsModel(feature_columns)
model.compile(optimizer='adam',
              loss='binary_crossentropy',
              metrics=['accuracy'],)

model.fit(train_ds,
          validation_data=val_ds,
          epochs=5)
```

**Other info / logs**
```bash
Epoch 1/5
---------------------------------------------------------------------------
InvalidArgumentError                      Traceback (most recent call last)
<ipython-input-27-fa952c9e276b> in <module>()
     21 model.fit(train_ds,
     22           validation_data=val_ds,
---> 23           epochs=5)

9 frames
/tensorflow-2.0.0b1/python3.6/tensorflow/python/keras/engine/training.py in fit(self, x, y, batch_size, epochs, verbose, callbacks, validation_split, validation_data, shuffle, class_weight, sample_weight, initial_epoch, steps_per_epoch, validation_steps, validation_freq, max_queue_size, workers, use_multiprocessing, **kwargs)
    641         max_queue_size=max_queue_size,
    642         workers=workers,
--> 643         use_multiprocessing=use_multiprocessing)
    644 
    645   def evaluate(self,

/tensorflow-2.0.0b1/python3.6/tensorflow/python/keras/engine/training_generator.py in fit(self, model, x, y, batch_size, epochs, verbose, callbacks, validation_split, validation_data, shuffle, class_weight, sample_weight, initial_epoch, steps_per_epoch, validation_steps, validation_freq, **kwargs)
    692         shuffle=shuffle,
    693         initial_epoch=initial_epoch,
--> 694         steps_name='steps_per_epoch')
    695 
    696   def evaluate(self,

/tensorflow-2.0.0b1/python3.6/tensorflow/python/keras/engine/training_generator.py in model_iteration(model, data, steps_per_epoch, epochs, verbose, callbacks, validation_data, validation_steps, validation_freq, class_weight, max_queue_size, workers, use_multiprocessing, shuffle, initial_epoch, mode, batch_size, steps_name, **kwargs)
    262 
    263       is_deferred = not model._is_compiled
--> 264       batch_outs = batch_function(*batch_data)
    265       if not isinstance(batch_outs, list):
    266         batch_outs = [batch_outs]

/tensorflow-2.0.0b1/python3.6/tensorflow/python/keras/engine/training.py in train_on_batch(self, x, y, sample_weight, class_weight, reset_metrics)
    916       self._update_sample_weight_modes(sample_weights=sample_weights)
    917       self._make_train_function()
--> 918       outputs = self.train_function(ins)  # pylint: disable=not-callable
    919 
    920     if reset_metrics:

/tensorflow-2.0.0b1/python3.6/tensorflow/python/keras/backend.py in __call__(self, inputs)
   3508         value = math_ops.cast(value, tensor.dtype)
   3509       converted_inputs.append(value)
-> 3510     outputs = self._graph_fn(*converted_inputs)
   3511 
   3512     # EagerTensor.numpy() will often make a copy to ensure memory safety.

/tensorflow-2.0.0b1/python3.6/tensorflow/python/eager/function.py in __call__(self, *args, **kwargs)
    570       raise TypeError("Keyword arguments {} unknown. Expected {}.".format(
    571           list(kwargs.keys()), list(self._arg_keywords)))
--> 572     return self._call_flat(args)
    573 
    574   def _filtered_call(self, args, kwargs):

/tensorflow-2.0.0b1/python3.6/tensorflow/python/eager/function.py in _call_flat(self, args)
    669     # Only need to override the gradient in graph mode and when we have outputs.
    670     if context.executing_eagerly() or not self.outputs:
--> 671       outputs = self._inference_function.call(ctx, args)
    672     else:
    673       self._register_gradient()

/tensorflow-2.0.0b1/python3.6/tensorflow/python/eager/function.py in call(self, ctx, args)
    443             attrs=("executor_type", executor_type,
    444                    "config_proto", config),
--> 445             ctx=ctx)
    446       # Replace empty list with None
    447       outputs = outputs or None

/tensorflow-2.0.0b1/python3.6/tensorflow/python/eager/execute.py in quick_execute(op_name, num_outputs, inputs, attrs, ctx, name)
     65     else:
     66       message = e.message
---> 67     six.raise_from(core._status_to_exception(e.code, message), None)
     68   except TypeError as e:
     69     if any(ops._is_keras_symbolic_tensor(x) for x in inputs):

/usr/local/lib/python3.6/dist-packages/six.py in raise_from(value, from_value)

InvalidArgumentError:  Expected D2 of index to be 2 got 3 at position 1
	 [[node feature_column_ds_model/dense_features_7/age_bucketized_X_thal_indicator/SparseCross (defined at <ipython-input-27-fa952c9e276b>:23) ]] [Op:__inference_keras_scratch_graph_32547]

Function call stack:
keras_scratch_graph
```

**Comment by ravikyram** [comment]
@Guzzii 
Is it possible for you to try latest TF versions (!pip install tf-nightly-2.0-preview==2.0.0.dev20190818)and let us know whether the issue persists? There were lots of performance improvements in the latest versions. I am not seeing the mentioned error in nightly versions.Thanks!

**Comment by Guzzii** [comment]
Thanks for the reply. I am trying to see if the error still exists in the latest version. The following code block still doesn't import the tf-nightly.
```python
try:
    # %tensorflow_version only exists in Colab.
    %tensorflow_version 2.x
except Exception:
    pass

import tensorflow as tf
```
Any suggestions on how to import the nightly build? Thanks

**Comment by ravikyram** [comment]
@Guzzii 
I tried with nightly versions ('2.0.0-dev20190819') in colab and i am not seeing any error. I am able to reproduce the issue with TF 2.0.0-beta1 version. Please install  (`!pip install tf-nightly-2.0-preview==2.0.0.dev20190819)`and try again and let me know if the issue still persists. Thanks!
![image](https://user-images.githubusercontent.com/51902062/63321485-83e93c00-c33e-11e9-9332-ac3d09d4ee46.png)

**Comment by yashk2810** [comment]
This should be fixed with tf2 nightly.

**Comment by ravikyram** [comment]
@Guzzii 
Can you try tf2-nighly and let us know if the issue still persists. Thanks!

**Comment by Guzzii** [comment]
@ravikyram I confirm the problem is fixed in `tf-nightly-2.0-preview==2.0.0.dev20190819`.

**Comment by ravikyram** [comment]
I am closing the issue since the query is been resolved.Thanks!
